contracts/Linea_SpokePool.sol

```diff
@@ -0,0 +1,226 @@
+        IMessageService _l2MessageService,
+        ITokenBridge _l2TokenBridge,
+        IUSDCBridge _l2UsdcBridge,
+        address _crossDomainAdmin,
+        address _hubPool
+    ) public initializer {
+        __SpokePool_init(_initialDepositId, _crossDomainAdmin, _hubPool);
+        _setL2TokenBridge(_l2TokenBridge);
+        _setL2MessageService(_l2MessageService);
+        _setL2UsdcBridge(_l2UsdcBridge);
+    }
+
+    /****************************************
+     *    LINEA-SPECIFIC ADMIN FUNCTIONS    *
+     ****************************************/
+
+    /**
+     * @notice Change L2 token bridge address. Callable only by admin.
+     * @param _l2TokenBridge New address of L2 token bridge.
+     */
+    function setL2TokenBridge(ITokenBridge _l2TokenBridge) public onlyAdmin nonReentrant {
+        _setL2TokenBridge(_l2TokenBridge);
+    }
+
+    /**
+     * @notice Change L2 message service address. Callable only by admin.
+     * @param _l2MessageService New address of L2 message service.
+     */
+    function setL2MessageService(IMessageService _l2MessageService) public onlyAdmin nonReentrant {
+        _setL2MessageService(_l2MessageService);
+    }
+
+    /**
+     * @notice Change L2 USDC bridge address. Callable only by admin.
+     * @param _l2UsdcBridge New address of L2 USDC bridge.
+     */
+    function setL2UsdcBridge(IUSDCBridge _l2UsdcBridge) public onlyAdmin nonReentrant {
+        _setL2UsdcBridge(_l2UsdcBridge);
+    }
+
+    /**************************************
+     *        INTERNAL FUNCTIONS          *
+     **************************************/
+
+    /**
+     * @notice Wraps any ETH into WETH before executing base function. This is necessary because SpokePool receives
+     * ETH over the canonical token bridge instead of WETH.
+     */
+    function _preExecuteLeafHook(address l2TokenAddress) internal override {
+        if (l2TokenAddress == address(wrappedNativeToken)) _depositEthToWeth();
+    }
+
+    // Wrap any ETH owned by this contract so we can send expected L2 token to recipient. This is necessary because
+    // this SpokePool will receive ETH from the canonical token bridge instead of WETH. This may not be neccessary
+    // if ETH on Linea is treated as ETH and the fallback() function is triggered when this contract receives
+    // ETH. We will have to test this but this function for now allows the contract to safely convert all of its
+    // held ETH into WETH at the cost of higher gas costs.
+    function _depositEthToWeth() internal {
+        //slither-disable-next-line arbitrary-send-eth
+        if (address(this).balance > 0) wrappedNativeToken.deposit{ value: address(this).balance }();
+    }
+
+    function _bridgeTokensToHubPool(uint256 amountToReturn, address l2TokenAddress) internal override {
+        // Linea's L2 Canonical Message Service, requires a minimum fee to be set.
+        uint256 minFee = IMessageService(l2MessageService).minimumFeeInWei();
```

Comment:
Can we add an external method that returns the minder so dataworker can more easily know how much value it should send on each call?

Reply:
Sure will do!

Reply:
Added here https://github.com/across-protocol/contracts-v2/pull/382/commits/916f600560a9600f042de8e8c0f09e1d8e7a27b8. New deployment can be tested [here](https://goerli.lineascan.build/address/0xfa3DA25059F4ff59dA7566B58D3299dB8a04691F#readProxyContract)